### PR TITLE
dev: switch from autopep8 to yapf

### PR DESCRIPTION
**Pull request by dcramer**
autopep8 has a number of issues, including:

- it doesnt handle unicode in files (it ignores encoding)
- its extremely slow
- its code rewriting is based on regexp and is wrong (thus we had to disable a bunch of rules)

this will require another autoformatting pass as yapf is far more aggressive and includes a lot more additional things than just basic pep8 validation

### CODE REVIEW

**Comment by dcramer** [comment]
column-limit needs to be column_limit

also parallel seems broken?

```
$ time yapf -r --in-place src/sentry -p
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/Users/dcramer/.virtualenvs/sentry/lib/python2.7/site-packages/concurrent/futures/process.py", line 208, in _queue_management_worker
    result_item = result_queue.get(block=True)
  File "/usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/multiprocessing/queues.py", line 117, in get
    res = self._recv()
TypeError: ('__init__() takes exactly 5 arguments (2 given)', <class 'lib2to3.pgen2.parse.ParseError'>, ("bad input: type=8, value=u')', context=(u'\\n        ', (32, 8))",))
```

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Changes to build requirements</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by dcramer** [comment]
the above issue was from this line:

```python
        return any(
            request.access.has_team_scope(team, s)
            for s in allowed_scopes,
        )
```

https://github.com/google/yapf/issues/412

**Comment by dcramer** [comment]
i'm open to disabling the docstring extra line:

blank_line_before_class_docstring = true

**Comment by dcramer** [comment]
Another one:

```
                itertools.chain.from_iterable(
                    groups.get(group, []) for groups in six.itervalues(digest),
                ),
```

**Comment by dcramer** [comment]
i will put up a PR that shows what this will do, and then we can repeat yesterdays process :/

File: setup.py [hunk 1]
```diff
@@ -210,17 +202,12 @@ def run(self):
...
     classifiers=[
-        'Framework :: Django',
-        'Intended Audience :: Developers',
-        'Intended Audience :: System Administrators',
-        'Operating System :: POSIX :: Linux',
-        'Programming Language :: Python :: 2',
-        'Programming Language :: Python :: 2.7',
-        'Programming Language :: Python :: 2 :: Only',
-        'Topic :: Software Development'
+        'Framework :: Django', 'Intended Audience :: Developers',
```
Comment (dcramer) [inline setup.py]: we could disable yapf on this chunk, but tbqh i dont care that much

File: src/sentry/api/bases/team.py [hunk 2]
```diff
@@ -17,19 +17,16 @@ class TeamPermission(OrganizationPermission):
     }
 
     def has_object_permission(self, request, view, team):
-        result = super(TeamPermission, self).has_object_permission(
-            request, view, team.organization)
+        result = super(TeamPermission,
```
Comment (dcramer) [inline src/sentry/api/bases/team.py]: might be configuration to make this slightly nicer, but dont care as long as i dont have to think about choosing one way or the other

**Comment by dcramer** [comment]
https://github.com/getsentry/sentry/compare/yapf/example

**Comment by tkaemming** [comment]
Most of the results seem OK but [this one](https://github.com/getsentry/sentry/compare/dev/yapfexample#diff-8386a19f42e28bbfb37abf249fe4139fR200) is downright offensive

The way it breaks [single argument collections or list comprehensions into multiple lines](https://github.com/getsentry/sentry/compare/dev/yapfexample#diff-ec9caea86e4f8f68c1e2ab06547f2d7eR216) is also kinda gross

Ultimately, I don't really care as long as it's consistent and we can stop thinking about it

**Comment by dcramer** [comment]
@tkaemming i sort of agree, but im also of the mindset that as long as its readable and i dont need to have an opinion, im happy

that said, this config I'm using in zeus has worked better:

https://github.com/getsentry/zeus/blob/master/setup.cfg

**Pull request merged**